logger_block.py

```diff
@@ -33,11 +31,16 @@ def process_signals(self, signals):
             None
         """
         log_func = self._get_logger()
-        for s in signals:
-            try:
-                log_func(s)
-            except:
-                self.logger.error("Failed to log signal")
+
+        try:
+            log_func(signals)
+        except:
+            if len(signals) > 1:
+                self.logger.exception("Failed to log list of signals: {}"
+                                      .format(signals))
+            else:
+                self.logger.exception("Failed to log signal: {}"
+                                      .format(signals[0]))
```

Comment:
Lets not include the signals in these messages. Maybe `len` instead? Because if you think about it, the same reason why line 36 would raise an exception would probably happen in these loggers too.